### ISSUE TITLE
Odd behavior using DOCKER_API_VERSION env variable.

### ISSUE THREAD

**Issue by serdmanczyk** [docker/cli#613]
**Description**

I am attempting to use the cli to talk to an older daemon (newer docker client) -> (older docker daemon).  The `DOCKER_API_VERSION` environment variable doesn't seem to be working properly in the client.

`docker version` output:
```
Client:
 Version:      17.09.0-ce
 API version:  1.24 (downgraded from 1.23)
 Go version:   go1.8.3
 Git commit:   afdb6d4
 Built:        Tue Sep 26 22:40:09 2017
 OS/Arch:      darwin/amd64

Error response from daemon: client is newer than server (client API version: 1.24, server API version: 1.23)
```

Docker env variables:
```
DOCKER_HOST=tcp://<redacted>:2376
DOCKER_API_VERSION=1.23
DOCKER_MACHINE_NAME=<redacted>
DOCKER_TLS_VERIFY=1
DOCKER_CERT_PATH=/Users/<redacted>
```

The client's API version line is odd in particular :
`API version:  1.24 (downgraded from 1.23)`
It appears as though something is odd in the client's version machinery, and it is 'downgrading' _up_ (requested API version) -> (client binary API version).

The client binary is from Docker CE for Mac.

As a workaround I can use homebrew to install an older client binary (docker@1.17), which gives me the output for the daemon version (running on RHEL7):
```
...
Server version: 1.11.2
Server API version: 1.23
Go version (server): go1.5.4
Git commit (server): b9f10c9
OS/Arch (server): linux/amd64
```

**Steps to reproduce the issue:**
1. Install Docker CE for Mac 17.09.0-ce
2. Install daemon 1.11.2 on RHEL7
3. Attempt to perform CLI commands using DOCKER_API_VERSION to downgrade client.

**Describe the results you received:**

Received "client is newer than server" response.

**Describe the results you expected:**

Expected DOCKER_API_VERSION env variable to downgrade version client would use to communicate with server.

**Output of `docker version`:**
(using latest Docker CE for Mac (17.09.0-ce))
```
Client:
 Version:      17.09.0-ce
 API version:  1.24 (downgraded from 1.23)
 Go version:   go1.8.3
 Git commit:   afdb6d4
 Built:        Tue Sep 26 22:40:09 2017
 OS/Arch:      darwin/amd64
Error response from daemon: client is newer than server (client API version: 1.24, server API version: 1.23)
```
(using older version client installed with homebrew)
```
Client version: 1.7.1
Client API version: 1.19
Go version (client): go1.8
Git commit (client): 786b29d
OS/Arch (client): darwin/amd64
Server version: 1.11.2
Server API version: 1.23
Go version (server): go1.5.4
Git commit (server): b9f10c9
OS/Arch (server): linux/amd64
```
**Output of `docker info`:**
(using latest Docker CE for Mac (17.09.0-ce))
```
Error response from daemon: client is newer than server (client API version: 1.24, server API version: 1.23)
```
(using older version client installed with homebrew)
```
Containers: 3775
Images: 8
Storage Driver: devicemapper
 Pool Name: docker-253:1-524289-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: ext4
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 7.323 GB
 Data Space Total: 107.4 GB
 Data Space Available: 35.27 GB
 Metadata Space Used: 160.5 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 1.987 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.140-RHEL7 (2017-05-03)
Logging Driver: json-file
Kernel Version: 3.10.0-693.2.2.el7.x86_64
Operating System: Red Hat Enterprise Linux Server 7.4 (Maipo)
CPUs: 2
Total Memory: 7.797 GiB
Name:<redacted>
ID: 6SJ6:WIXJ:35MM:RZIB:MWIS:K7QZ:TKYJ:D7KM:4ODW:PPFC:LYAV:ZZQ2
```

**Comment by dnephin** [comment]
If I remember correctly version downgrading was only supported starting at version v1.24. I think what is happening is that the server is returning an error when the client attempts to ping it to get the API information. Since the server errored the client defaults to version v1.24, hence the weird output.

**Comment by thaJeztah** [comment]
This may be related to a fix we recently merged /cc @tonistiigi

**Comment by thaJeztah** [comment]
Was thinking of this one (but not near a computer to check https://github.com/moby/moby/pull/35008)

**Comment by thaJeztah** [comment]
> Any idea when that will be rolled into the stable release of Docker CE for mac? As soon as it's out I can retest myself and close this issue if it works.

Updates to stable releases are usually done if a `P0` ("show stopper / critical issue (such as data-loss etc") is found; at this moment there's no such issue (to my knowledge), but it may be included when a patch release is done for stable. Otherwise, it'll end up in the next stable release.

Given that this issue is resolved in "master", and in docker 17.10 and up, and that a workaround is possible, I'm closing this issue for housekeeping, but feel free to comment 👍

**Comment by serdmanczyk** [comment]
Thanks!  I've switched to the edge release for now and that appears to be working.
